exla/lib/exla/defn.ex

```diff
@@ -56,69 +56,227 @@ defmodule EXLA.Defn do
     {expr, acc}
   end
 
+  @doc false
+  def __stream__(key, input, acc, vars, fun, options) do
+    {run_options, compile_options} = Keyword.pop(options, :run_options, [])
+    keep_on_device? = Keyword.get(run_options, :keep_on_device, false)
+    run_options = Keyword.put(run_options, :keep_on_device, true)
+
+    # The input vars should not be converted to buffers as they come from infeed
+    input_vars = Nx.Defn.Tree.flatten_list([input])
+    input_shape = EXLA.Shape.make_tuple_shape(Enum.map(input_vars, &nx_to_shape/1))
+    acc_vars = Nx.Defn.Tree.flatten_list([acc])
+    split_fun = &split_stream(&1, &2, length(input_vars), length(acc_vars))
+
+    comp_fun =
+      &to_stream_computation(key, input_vars, input_shape, acc_vars, &1, &2, compile_options)
```

Comment:
perhaps `fn ->` syntax here would be more clear on which two arguments this function expects